src/Nethermind/Nethermind.Synchronization/Blocks/BlockDownloadContext.cs

```diff
@@ -129,6 +130,12 @@ public bool TrySetReceipts(int index, TxReceipt[]? receipts, out Block block)
             return result;
         }
 
+        public Block GetBlockByRequestIdx(int index)
```

Comment:
why did we change pre merge block downloader? Could you elaborate?

Reply:
- Fixed an issue where if the block downloader downloads bodies from peer where none of the header hash was available, the peer will response with zero bodies which triggers an infinite loop. Fixes by have block downloader stop downloading part way and only insert until it found a block without a body when it should have one.
- Make sure MergeBlockDownloader does not request block higher than what the peer have.

This is for the first fix where we need the download receipt to stop validating once it detect that the block download failed for this block.

Reply:
Moving the change to another pr